cpp/source/deal.II/geometry.templates.h

```diff
@@ -158,6 +158,7 @@ unsigned int Geometry<dim>::compute_cell_weight(
   if (std::find(collector.begin(), collector.end(), material) !=
       collector.end())
     return weights[internal::WEIGHT_TYPE::collector];
+  return -1;
```

Comment:
Can you return ```dealii::numbers::invalid_unsigned_int``` to make it clear that we shouldn't go here? Then merge yourself.